networkx/algorithms/isomorphism/vf2pp.py

```diff
@@ -0,0 +1,250 @@
+"""
+***************
+VF2++ Algorithm
+***************
+
+An implementation of the VF2++ algorithm for Graph Isomorphism testing.
+
+The simplest interface to use this module is to call:
+`vf2pp_is_isomorphic`: to check whether two graphs are isomorphic.
+`vf2pp_mapping`: to obtain the node mapping between two graphs, in case they are isomorphic.
+`vf2pp_all_mappings`: to generate all possible mappings between two graphs, if isomorphic.
+
+"""
+import collections
+
+import networkx as nx
+from networkx.algorithms.isomorphism.vf2pp_helpers.candidates import _find_candidates
+from networkx.algorithms.isomorphism.vf2pp_helpers.feasibility import _feasibility
+from networkx.algorithms.isomorphism.vf2pp_helpers.node_ordering import _matching_order
+from networkx.algorithms.isomorphism.vf2pp_helpers.state import (
+    _restore_state,
+    _update_state,
+)
+
+__all__ = ["vf2pp_mapping", "vf2pp_is_isomorphic"]
+
+
+def vf2pp_mapping(G1, G2, node_labels=None, default_label=None):
+    """Return an isomorphic mapping between `G1` and `G2` if it exists.
+
+    Parameters
+    ----------
+    G1,G2: NetworkX Graph or MultiGraph instances.
+        The two graphs to check for isomorphism or monomorphism.
+
+    node_labels: Label name
+        The label name of all nodes
+
+    default_label: Label name
+        Let the user pick a default label value
+
+    Returns
+    -------
+    Node mapping, if the two graphs are isomorphic. None otherwise.
+    """
+    try:
+        mapping = next(vf2pp_all_mappings(G1, G2, node_labels, default_label))
+        return mapping
+    except StopIteration:
+        return None
+
+
+def vf2pp_is_isomorphic(G1, G2, node_labels=None, default_label=None):
+    """Examines whether G1 and G2 are isomorphic.
+
+    Parameters
+    ----------
+    G1,G2: NetworkX Graph or MultiGraph instances.
+        The two graphs to check for isomorphism or monomorphism.
+
+    node_labels: Label name
+        The label name of all nodes
+
+    default_label: Label name
+        Let the user pick a default label value
+
+    Returns
+    -------
+    True if the two graphs are isomorphic. False otherwise.
+    """
+    return True and not (not vf2pp_mapping(G1, G2, node_labels, default_label))
```

Comment:
Did we want the `True and not (not stuff)` syntax here? Is there a more readable version?

Reply:
Yeah this is definitely too hard to grok - I vote for simpler logic!